pkg/command/root.go

```diff
@@ -240,8 +247,41 @@ func newRootCmd() *cobra.Command {
 		// silencing usage for now as we are getting double usage from plugins on errors
 		SilenceUsage: true,
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			// Sets the verbosity of the logger if TANZU_CLI_LOG_LEVEL is set
-			setLoggerVerbosity()
+			// Manually parse the flags
+			for i := 0; i < len(args); i++ {
+				arg := args[i]
+				if arg == "-v" || arg == "--verbose" {
+					if i+1 < len(args) {
+						nextArg := args[i+1]
+						// Check if the next argument is another flag
+						if strings.HasPrefix(nextArg, "-") {
+							log.Errorf("Missing value for verbose flag")
+						}
+						// Try to convert the next argument to an integer
+						if v, err := strconv.Atoi(nextArg); err == nil {
+							verbose = v
+						} else {
+							log.Errorf("Invalid value for verbose flag: %v\n", err)
+						}
+						// Skip the next argument
+						i++
+					} else {
+						log.Errorf("Missing value for verbose flag")
+					}
+				}
+			}
+
+			// Sets the verbosity of the logger
```

Comment:
We still need to check if a plugin has also declared a `--verbose` flag, and in that case pass it to the plugin.
I now realize this is a little complicated because if the user does:
```
tz telemetry --verbose true
```
the command will fail because Cobra will try to parse `true` as an integer.

We can do three things:
1. make the `--verbose` flag a string so that it never fails to parse and handle the value checking ourselves.  But that won't work if the plugin set the flag as a boolean since it does not take a value while a string flag does.  We may be able to get around that by setting the flags `NoOptDefVal` field.  I haven't tried.
2. _Before_ calling `Execute()` which means before Cobra parses the flags, manually parse `os.Args` to find `--verbose` and remove it from the args if the plugin has declared it.  That means also figuring out which plugin is being called.  This gets pretty tricky but can be done.  I would prefer to avoid it though.
3. Discuss if we can work with plugin authors to figure out if we really need to worry about plugins having declared `--verbose` as something else than `int` or `string`.

Reply:
For Plugin commands - CLI doesnt parse the args to its flags but sends all the args to plugin as arguments. we only verify if verbose flag is passed by user and then set the verbosity accordingly.  
Even if verbose flag is bool/string for plugins, CLI would pass those args.  As part of verbosity we need to determine what should we set the verbosity if verbose is true or false.